api/v1alpha2/cluster_types.go

```diff
@@ -42,28 +42,33 @@ type ClusterSpec struct {
 
 /// [ClusterSpec]
 
-/// [ClusterNetworkingConfig]
+/// [ClusterNetworking]
 // ClusterNetworkingConfig specifies the different networking
 // parameters for a cluster.
-type ClusterNetworkingConfig struct {
+type ClusterNetworking struct {
+	// APIServerPort specifies the port the API Server should bind to.
+	// Defaults to 6443.
+	// +optional
+	APIServerPort *int32 `json:"apiServerPort,omitempty"`
+
 	// The network ranges from which service VIPs are allocated.
 	// +optional
-	Services *NetworkRanges `json:"services,omitempty"`
+	Services *NetworkRange `json:"services,omitempty"`
 
 	// The network ranges from which Pod networks are allocated.
 	// +optional
-	Pods *NetworkRanges `json:"pods,omitempty"`
+	Pods *NetworkRange `json:"pods,omitempty"`
 
 	// Domain name for services.
 	// +optional
 	ServiceDomain string `json:"serviceDomain,omitempty"`
 }
 
-/// [ClusterNetworkingConfig]
+/// [ClusterNetworking]
 
 /// [NetworkRanges]
 // NetworkRanges represents ranges of network addresses.
-type NetworkRanges struct {
+type NetworkRange struct {
```

Comment:
I think pluralized is fine since it could in theory contain multiple network ranges rather than the single network range we generally use today.

Reply:
I renamed this because the field containing the blocks is already plural and can achieve what you're describing

Reply:
Calling it NetworkRange would lead me to assume that it would only accept one CIDR rather than the multiple it does

Reply:
Renamed it back, don't care much about it tbh 😂